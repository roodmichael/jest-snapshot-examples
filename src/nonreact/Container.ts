export class Container {

    visibility: string;

    constructor(directive : any) {
        this.visibility = directive.visibility;
    }

    render() : HTMLElement {
        let container : HTMLDivElement = document.createElement("div");
        let text : Text = document.createTextNode("text inside div");

        container.appendChild(text);

        container.style.cssText = `visibility:${this.visibility};`;
        return container;
    }
}
