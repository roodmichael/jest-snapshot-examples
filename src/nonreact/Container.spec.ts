import { Container } from './Container';

const templateVisible = {
    visibility: 'visible'
};

const templateHidden = {
    visibility: 'hidden'
};

test('renders a container', () => {
    const result = new Container(templateVisible).render();
    expect(result).toMatchSnapshot();
});

test('visibility is hidden', () => {
    const result = new Container(templateHidden).render();
    expect(result).toMatchSnapshot();
});
