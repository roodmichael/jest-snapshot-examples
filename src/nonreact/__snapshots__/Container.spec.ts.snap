// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a container 1`] = `
<div
  style="visibility: visible;"
>
  text inside div
</div>
`;

exports[`visibility is hidden 1`] = `
<div
  style="visibility: hidden;"
>
  text inside div
</div>
`;
